src/construction.cpp

```diff
@@ -1067,10 +1073,21 @@ void complete_construction( player *p )
     }
     // Make the terrain change
     if( !built.post_terrain.empty() ) {
+        const tripoint terp_actual = built.post_flags.count( "is_roof" ) ? terp + tripoint_above : terp;
         if( built.post_is_furniture ) {
-            here.furn_set( terp, furn_str_id( built.post_terrain ) );
+            here.furn_set( terp_actual, furn_str_id( built.post_terrain ) );
         } else {
-            here.ter_set( terp, ter_str_id( built.post_terrain ) );
+            here.ter_set( terp_actual, ter_str_id( built.post_terrain ) );
+            // Make a roof if constructed terrain should have it and it's an open air
+            if( construct::check_up_OK( terp ) ) {
+                const int_id<ter_t> post_terrain = ter_id( built.post_terrain );
+                if( post_terrain->roof ) {
+                    const tripoint top = terp + tripoint_above;
+                    if( here.ter( top ) == t_open_air ) {
```

Comment:
Can you implement this is a way that doesn't check a single hardcoded terrain id? e.g. a flag or similar?